src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DacFxRequest.cs

```diff
@@ -0,0 +1,34 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+using Microsoft.SqlTools.ServiceLayer.TaskServices;
+using Microsoft.SqlTools.ServiceLayer.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.DacFx.Contracts
+{
+    /// <summary>
+    /// Parameters for a DacFx request.
+    /// </summary>
+    public interface IDacFxParams : IScriptableRequestParams
```

Comment:
Seems like we can just make this a class and do away with all of the IDacFxParams implementations. But, this abstraction will break down in the future since, each operations will have operation specific parameters. This works for now, but think about how we can address this in the future.

Reply:
Changed it to be a base class instead